exporter/otlpexporter/otlp.go

```diff
@@ -141,38 +141,49 @@ func processError(err error) error {
 		return nil
 	}
 
-	// Now, this is this a real error.
+	// Now, this is a real error.
 
-	if !shouldRetry(st.Code()) {
-		// It is not a retryable error, we should not retry.
+	// Check if error is permanent, in which case we shouldn't ever retry.
```

Comment:
I have a hard-time reading the logic in lines 147-163.

Can we refactor this to avoid using a pair of shouldNeverRetry/shouldAlwaysRetry functions and instead show the entirety of the logic in one place?

Presumably all we need to do is modify the shouldRetry function and modify its behavior for the ResourceExhausted case. No other changes should be necessary if [this is correct](https://github.com/open-telemetry/opentelemetry-collector/pull/5147#issuecomment-1091952940).

Reply:
Alright, gonna make it simpler